src/graph.rs

```diff
@@ -159,33 +158,87 @@ impl Range {
   }
 }
 
-#[derive(Debug, Clone)]
-pub enum ModuleError {
```

Comment:
I split this up into way more targeted error classes so that I could more easily tell when an integrity check fails. This will make things easier in the CLI.